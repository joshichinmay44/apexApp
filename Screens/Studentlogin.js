import React from 'react';
import {Appbar, Button, TextInput} from 'react-native-paper';
import {
  View,
  Text,
  ScrollView,
  StyleSheet,
  TouchableOpacity,
  Image,
  SafeAreaView,
  ActivityIndicator,
} from 'react-native';
import styles from '../style/StyleLogin';
import * as firebase from 'firebase';

export default class Studentlogin extends React.Component {
  static navigationOptions = {
    headerShown: false,
  };

  state = {
    username: '',
    password: '',
  };

  static navigationOptions = {
    headerShown: false,
  };
  state = {
    username: '',
    password: '',
    email: '',
    status: '',
    Students: [],
    Teachers: [],
    active: 0,
    indicator: false,
    
  };

  validate=()=>{
    let user=0
    const username = this.state.username;

    let ex
  //  console.log(username)
   const myitems=firebase.database().ref('Students/')
    myitems.on('value', datasnap => {
      if (datasnap.val()) {
        this.setState({Students: Object.values(datasnap.val())}, () => {
          console.log('validating stud');
          const ex = this.state.Students;
          console.log(ex);
          Object.keys(ex).map(function(key) {
            if (ex[key].match(username)) {
              console.log('Entered Student')
              user=1;
              
           
            }
          }
          )
            }
          )
            }
            }
          )
 
          const myitemsTeach=firebase.database().ref('Teachers/')
          myitemsTeach.on('value', datasnap => {
            if (datasnap.val()) {
              this.setState({Teachers: Object.values(datasnap.val())}, () => {
                console.log('validating teach ');
                 ex = this.state.Teachers;
                console.log(ex);
                Object.keys(ex).map(function(key) {
                  if (ex[key].match(username)) {
                    console.log('Entered Teacher')
                    return true
                      
                  }
                }
                )

                  }
                )
                  }
                  }
                )


                  

            }
  



  onLoginSuccess = () => {
    this.setState({username: '', password: ''});
    this.load();
  };
  studentProfile = () => {
    this.setState({indicator: true});
    
    const username = this.state.username;
    console.log('in function');
    //console.log(this.validate)

   // console.log(decision)
    if(true)

{     

    firebase
      .auth()
      .signInWithEmailAndPassword(this.state.username, this.state.password)
      .then(() => {
        console.log('successfully logged in');
        let activeState = 0;
        const myitems = firebase.database().ref('Students/');
        myitems.on('value', datasnap => {
          if (datasnap.val()) {
            this.setState({Students: Object.values(datasnap.val())}, () => {
              console.log('fetching stud');
              const ex = this.state.Students;
              console.log(ex);
              Object.keys(ex).map(function(key) {
                if (ex[key].match(username)) {
                  activeState = 1;
                  userFound=1;
                }
                

              });
              if (activeState == 1) {
                activeState = 0;
                this.props.navigation.navigate('StudentProfile', {
                  username: username,
                });
              }
             
              console.log('wait');
             
              
            });
          }
        });
        const myitem = firebase.database().ref('Teachers/');
        myitem.on('value', datasnap => {
          if (datasnap.val()) {
            this.setState({Teachers: Object.values(datasnap.val())}, () => {
              console.log('fetching teacher');
              const exx = this.state.Teachers;
              console.log(exx);
              Object.keys(exx).map(function(key) {
                if (exx[key].match(username)) {
                  activeState = 2;
                  
                }
              });
              if (activeState == 2) {
                activeState = 0;
                this.props.navigation.navigate('TeacherProfile', {
                  username: username,
                });
              }

              
            });
          }
        });
          
      

           console.log('wait')
             
        

      })
      .then(
        
        this.onLoginSuccess)
        
      .catch(function(error) {
       
        var errorCode = error.code;
        var errorMessage = error.message;
        alert(errorMessage);
      })
     
      .then(this.onLoginSuccess);
    }
  };
  load() {
    this.setState({indicator: false});
  }
  render() {
    return (
      <View style={styles.Container}>
        <Appbar.Header>
          <Appbar.Action
            icon="menu"
            onPress={() => this.props.navigation.openDrawer()}
          />
          <Appbar.Content title="Login" />
        </Appbar.Header>
        <ScrollView style={styles.scrollView}>
          <View style={styles.body}>
            <View style={styles.imageview}>
              <Image
                style={styles.userimage}
                source={require('../images/ApexLogo.jpg')}
              />
            </View>

            <View style={styles.textinputview}>
              <TextInput
                label="Username"
                mode="outlined"
                value={this.state.text}
                style={styles.mytextinput}
                onChangeText={username => this.setState({username})}
                onFocus={() => this.setState({username: ''})}
                value={this.state.username}
              />

              <TextInput
                label="Password"
                mode="outlined"
                value={this.state.text}
                secureTextEntry={true}
                style={styles.mytextinput}
                onChangeText={password => this.setState({password})}
                onFocus={() => this.setState({password: ''})}
                value={this.state.password}
              />
              <View style={styles.button}>
                {this.state.indicator ? (
                  <ActivityIndicator
                    size="large"
                    color="#084d7b"
                    style={styles.indicator}
                  />
                ) : null}
                {!this.state.indicator ? (
                  <Button mode="contained" onPress={this.studentProfile}>
                    {' '}
                    Login{' '}
                  </Button>
                ) : null}
              </View>
            </View>
          </View>
        </ScrollView>
      </View>
    );
  }
}
