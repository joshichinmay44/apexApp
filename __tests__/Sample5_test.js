import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Studentlogin from '../Screens/Studentlogin';
import {TestScheduler} from 'jest';

/* global.fetch = jest.fn(() => new Promise(resolve => resolve()));
jest.mock('react-native-gesture-handler', () => {});
jest.mock('react-native-reanimated', () => {});
jest.mock('@react-navigation/drawer', () => {});
jest.mock('@react-navigation/stack', () => {});
 */
test('renders correctly', () => {
  const snap = renderer.create(<Studentlogin />).toJSON();
  expect(snap).toMatchSnapshot();
  let loginComponent = renderer.create(<Studentlogin />).getInstance();
  loginComponent.load();
  loginComponent.onLoginSuccess();
});
