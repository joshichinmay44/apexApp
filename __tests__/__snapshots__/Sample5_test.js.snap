// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fdfefe",
      "height": "100%",
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#6200ee",
          "elevation": 4,
          "zIndex": 0,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#6200ee",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": 0,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        <View
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                Ôçú
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
              "paddingHorizontal": 12,
            },
            Array [
              Object {
                "marginLeft": 8,
              },
              Object {
                "alignItems": "center",
              },
              undefined,
            ],
          ]
        }
      >
        <Text
          accessibilityRole="header"
          accessibilityTraits="header"
          accessible={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "fontSize": 17,
                },
                undefined,
              ],
            ]
          }
        >
          Login
        </Text>
      </View>
      <View
        style={
          Object {
            "width": 48,
          }
        }
      />
    </View>
  </RCTSafeAreaView>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flexDirection": "column",
            "height": "100%",
            "paddingBottom": "50%",
          }
        }
      >
        <View>
          <Image
            source={
              Object {
                "testUri": "../../../images/ApexLogo.jpg",
              }
            }
            style={
              Object {
                "height": "70%",
                "marginLeft": "25%",
                "marginTop": "10%",
                "width": "50%",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "50%",
              "padding": "5%",
            }
          }
        >
          <View
            style={
              Array [
                undefined,
                Object {
                  "margin": "3%",
                  "width": "100%",
                },
              ]
            }
          >
            <View>
              <View
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                    },
                    Object {
                      "backgroundColor": "#f6f6f6",
                      "borderColor": "rgba(0, 0, 0, 0.54)",
                      "borderRadius": 4,
                      "borderWidth": 1,
                    },
                  ]
                }
              />
              <View
                style={
                  Object {
                    "minHeight": 64,
                    "paddingBottom": 0,
                    "paddingTop": 8,
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      Object {
                        "backgroundColor": "#f6f6f6",
                        "bottom": 0,
                        "height": 2,
                        "left": 10,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 6,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                        ],
                        "width": 8,
                      }
                    }
                  />
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "backgroundColor": "#f6f6f6",
                        "color": "transparent",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 18,
                        "opacity": 0,
                        "paddingHorizontal": 0,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 37,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                          Object {
                            "scaleY": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    Username
                  </Text>
                  <Text
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      Object {
                        "color": "#6200ee",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 0,
                        "paddingHorizontal": 14,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 36,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    Username
                  </Text>
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 1,
                        "paddingHorizontal": 14,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 36,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    Username
                  </Text>
                </View>
                <TextInput
                  allowFontScaling={true}
                  editable={true}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder=""
                  placeholderTextColor="rgba(0, 0, 0, 0.54)"
                  rejectResponderTermination={true}
                  selectionColor="#6200ee"
                  style={
                    Array [
                      Object {
                        "flexGrow": 1,
                        "margin": 0,
                        "paddingHorizontal": 14,
                        "textAlign": "left",
                        "zIndex": 1,
                      },
                      Object {
                        "height": 56,
                      },
                      Object {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "textAlignVertical": "center",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                undefined,
                Object {
                  "margin": "3%",
                  "width": "100%",
                },
              ]
            }
          >
            <View>
              <View
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                    },
                    Object {
                      "backgroundColor": "#f6f6f6",
                      "borderColor": "rgba(0, 0, 0, 0.54)",
                      "borderRadius": 4,
                      "borderWidth": 1,
                    },
                  ]
                }
              />
              <View
                style={
                  Object {
                    "minHeight": 64,
                    "paddingBottom": 0,
                    "paddingTop": 8,
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      Object {
                        "backgroundColor": "#f6f6f6",
                        "bottom": 0,
                        "height": 2,
                        "left": 10,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 6,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                        ],
                        "width": 8,
                      }
                    }
                  />
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "backgroundColor": "#f6f6f6",
                        "color": "transparent",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 18,
                        "opacity": 0,
                        "paddingHorizontal": 0,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 37,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                          Object {
                            "scaleY": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    Password
                  </Text>
                  <Text
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      Object {
                        "color": "#6200ee",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 0,
                        "paddingHorizontal": 14,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 36,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    Password
                  </Text>
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 1,
                        "paddingHorizontal": 14,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 36,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    Password
                  </Text>
                </View>
                <TextInput
                  allowFontScaling={true}
                  editable={true}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder=""
                  placeholderTextColor="rgba(0, 0, 0, 0.54)"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  selectionColor="#6200ee"
                  style={
                    Array [
                      Object {
                        "flexGrow": 1,
                        "margin": 0,
                        "paddingHorizontal": 14,
                        "textAlign": "left",
                        "zIndex": 1,
                      },
                      Object {
                        "height": 56,
                      },
                      Object {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "textAlignVertical": "center",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "margin": "5%",
                "width": "60%",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#6200ee",
                  "borderColor": "transparent",
                  "borderRadius": 4,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "elevation": 2,
                  "minWidth": 64,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 0.75,
                    "width": 0,
                  },
                  "shadowOpacity": 0.24,
                  "shadowRadius": 1.5,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityStates={Array []}
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 4,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "400",
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "letterSpacing": 1,
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          undefined,
                          Object {
                            "textTransform": "uppercase",
                          },
                          Object {
                            "color": "#ffffff",
                            "fontFamily": "System",
                            "fontWeight": "500",
                          },
                          Object {
                            "fontFamily": "System",
                            "fontWeight": "500",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                     
                    Login
                     
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
